### ISSUE TITLE
Interpreter and ecoreref code generation

### ISSUE THREAD

**Issue by LorenzoBettini** [LorenzoBettini/edelta#96]
This one works at runtime

```
modifyEcore introduceAbstractArc epackage petrinet {
	addNewEClass("Arc") [
		abstract = true
		val List<EAttribute> attributes = #[ecoreref(PTArc.weight), ecoreref(TPArc.weight)]
//		refactorings.extractSuperclass(it, attributes);
	]
}
```

generating

```java
final List<EAttribute> attributes = ...newArrayList(
getEAttribute("petrinet", "PTArc", "weight"), getEAttribute("petrinet", "TPArc", "weight")));
```

while this one

```
modifyEcore introduceAbstractArc epackage petrinet {
	addNewEClass("Arc") [
		abstract = true
		val List<EAttribute> attributes = #[ecoreref(PTArc.weight), ecoreref(TPArc.weight)]
		refactorings.extractSuperclass(it, attributes);
	]
}
```

leads to wrong generated code

```java
final List<EAttribute> attributes = ...newArrayList(
getEAttribute("petrinet", "Arc", "weight"), getEAttribute("", "", "weight")));
```

Since ```refactorings.extractSuperclass(it, attributes)``` lifts the attributes in the superclass

So the interpreter breaks code generation.

That's because the last `extractSuperclass` removes the passed attributes from the classes `PTArc` and `TPArc`, thus when the generator accesses those Ecore elements they are not there anymore.

During the interpretation we should record ecoreref's argument information so that the generator can use the original one. Something similar to what we do with the original element, but with the new syntax that does not work anymore.

This new strategy should also enable us to catch accesses to element that are not there anymore (open a new issue?)

**Comment by LorenzoBettini** [comment]
The other issue https://github.com/LorenzoBettini/edelta/issues/93 should be fixed first

**Comment by LorenzoBettini** [comment]
We could store information in the ecoreref model:

- String type (e.g., EClass, EAttribute, etc.) corresponding to type.simpleName
- EPackage package
- EClassifier classifier (can be null when referring to EPackage only)
- ENamedElement element (can be null when referring to EClassifier only)

**Comment by LorenzoBettini** [comment]
Closed by #101
